services/src/main/java/org/openlearn/web/rest/CourseResource.java

```diff
@@ -148,14 +150,45 @@ public CourseResource(final CourseService courseService, final StudentCourseServ
 	 */
 	@GetMapping("/courses/{id}/students")
 	@Timed
-	public ResponseEntity<List<CourseStudent>> studentsInCourse(@PathVariable final Long id, @ApiParam final Pageable pageable) {
+	public ResponseEntity<List<CourseStudentDTO>> studentsInCourse(@PathVariable final Long id, @ApiParam final Pageable pageable) {
 		log.debug("REST request to get students in course with id {}", id);
-		final Page<CourseStudent> page = studentCourseService.findByCourseId(id, pageable);
+		final Page<CourseStudentDTO> page = studentCourseService.findByCourseId(id, pageable);
 		final HttpHeaders headers = PaginationUtil.generatePaginationHttpHeaders(page, "/api/courses"+id+"/students");
 		return new ResponseEntity<>(page.getContent(), headers, HttpStatus.OK);
 	}
 
 	/**
+	 * Get students for course
+	 *
+	 * @param id the id of the course to add the student to
+	 * @return the students in the course
+	 */
+	@GetMapping("/courses/{id}/coursesByStudent")
+	@Timed
+	public ResponseEntity<List<Course>> coursesByStudent(@PathVariable final Long id, @ApiParam final Pageable pageable) {
+		log.debug("REST request to get students in course with id {}", id);
+		final Page<Course> page = courseService.findAllByStudentId(pageable,id);
+		final HttpHeaders headers = PaginationUtil.generatePaginationHttpHeaders(page, "/api/courses/"+id+"/coursesByStudent");
+		return new ResponseEntity<>(page.getContent(), headers, HttpStatus.OK);
+	}
+
+	/**
+	 * Get students not in course
+	 *
+	 * @param id the id of the course to search
+	 * @return the students not in the course
+	 */
+	@GetMapping("/courses/{id}/studentsNot")
+	@Timed
+	public ResponseEntity<List<User>> studentsNotInCourse(@PathVariable final Long id) {
+		log.debug("REST request to get students not in course with id {}", id);
+		final List<User> students = studentCourseService.findByCourseIdNot(id);
+		//Left over from pagination removal
```

Comment:
I couldn't get it to work without the native and the native doesn't allow pagination. I can look again and see if i can fix it/ add it in.